test/dynamo/test_install_params_as_graph_attr.py

```diff
@@ -0,0 +1,156 @@
+# Owner(s): ["module: dynamo"]
+from typing import Callable
+
+import torch
+import torch._dynamo
+import torch._dynamo.test_case
+import torch._dynamo.testing
+from torch._dynamo.testing import EagerAndRecordGraphs
+from torch.fx.graph_module import GraphModule
+
+
+def compile_and_extract_graph(
+    fn, *args, **kwargs
+) -> tuple[Callable, list[torch.fx.GraphModule]]:
+    backend = EagerAndRecordGraphs()
+    result_fn = torch.compile(backend=backend)(fn)
+    # Run fn to capture graph
+    _ = result_fn(*args, **kwargs)
+    return result_fn, backend.graphs
+
+
+def get_num_input_nodes(graph: GraphModule) -> int:
+    """Returns the number of input nodes in the input GraphModule
+
+    NOTE: Assumes a bijection between input nodes and placeholders
+    but, this is not always the case as symbolic tracing can introduce
+    additional placeholders.
+
+    For these test cases, there is no symbolic tracing, so this is
+    a safe assumption
+    """
+    placeholder_cnt = 0
+    for node in graph.graph.nodes:
+        if node.op == "placeholder":
+            placeholder_cnt += 1
+    return placeholder_cnt
+
+
+class SimpleLinearModule(torch.nn.Module):
+    """
+    Simple linear model with 1 parameter and 1 buffer
+    for basic testing purposes
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.fwd = torch.nn.Linear(5, 1)
+
+    def forward(self, x):
+        return self.fwd(x)
+
+
+class InstallParamsAsGraphAttrTests(torch._dynamo.test_case.TestCase):
+    def check_num_inputs_and_equality(
+        self, original_fn, example_input, expected_num_inputs: int
+    ) -> None:
+        """Compiles the original fn, then:
+        * Checks that the number of inputs in the graph is expected_num_inputs
+        * Checks that the compiled fn and original fn are equal
+        """
+        # Setup config
+        torch._dynamo.config.inline_inbuilt_nn_modules = True
+        # Currently a no-op
+        torch._dynamo.config.install_params_as_graph_attr = True
```

Comment:
Use `@torch._dynamo.config.patch(inline_inbuilt_nn_modules=False)` (grep for it)

Tests in CI are run in parallel, and these are global configs which can introduce flakiness.